src/block.rs

```diff
@@ -352,8 +458,35 @@ mod tests {
             "bcc6eb2ab6c97b9b4590825b9136f100b22e090c0469818572b8b93926a79f28"
         );
         assert_eq!(block.header.version, 0x20000000);
-        assert_eq!(block.header.proof.challenge.len(), 1 + 3 * 34 + 2);
-        assert_eq!(block.header.proof.solution.len(), 144);
+        if let ExtData::BlockProof(ref proof) = block.header.ext {
+            assert_eq!(proof.challenge.len(), 1 + 3 * 34 + 2);
+            assert_eq!(proof.solution.len(), 144);
+        } else {
+            panic!("dynafed block should not be dynafed");
```

Comment:
"Liquid v1-style static `signblockscript block" should not be dynafed ?

Reply:
How about "dynafed test vector was parsed as non-dynafed"?